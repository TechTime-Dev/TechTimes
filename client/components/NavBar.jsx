import React from "react";
import { Link } from "react-router-dom";

const NavBar = () => {
  return (
    <nav className="font-semibold max-w-[1536px] py-3 bg-[#e6f2f7] flex items-center justify-between mx-auto rounded-md">
      <div className="navbar-div flex items-center justify-between w-full pr-20 pl-20">
        <div className="brand flex items-center">
          <img
            src={require("../../public/logo-removebg.png").default}
            width="150"
            height="150"
          ></img>
          <Link to="/" className="text-black text-6xl font-josefin">
            Tech Times
          </Link>
        </div>
        <div classname="nav-div flex justify-between items-center">
          <Link
            to="/"
            className="text-black text-3xl hover:text-gray-300 font-ptsans"
          >
            News Feed
          </Link>
          <Link
            to="/favorite"
            className="text-black text-3xl hover:text-gray-300 font-ptsans ml-40"
          >
            Favorites
          </Link>
        </div>
        <div>
          <button className="bg-transparent text-black  rounded-md font-ptsans text-3xl ml-6">
            Log out
          </button>
        </div>
      </div>
    </nav>
  );
};

export default NavBar;
