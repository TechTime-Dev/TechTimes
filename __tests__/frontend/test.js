import React from 'react';
import { render, screen } from '@testing-library/react';
import NavBar from '../../client/components/NavBar'; // Assuming the navbar component is named NavBar
import ArticleCard from '../../client/components/articleCard';
import '@testing-library/jest-dom';

describe('NavBar', () => {
  it('Snapshot test', () => {
    expect(NavBar).toMatchSnapshot();
  });
});

describe('Market', () => {
  let text;
  const props = {
    title: 'test',
    author: 'test2',
  };

  beforeEach(() => {
    text = render(<ArticleCard {...props} />);
  });

  it('should display title and author', () => {
    expect(text.getByText('test')).toBeInTheDocument();
    expect(text.getByText('test2')).toBeInTheDocument();
  });
});
