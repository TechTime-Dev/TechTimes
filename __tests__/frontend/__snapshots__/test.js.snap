// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Snapshot test 1`] = `[Function]`;
